test/integration/acceptance/grant_permission_test.cpp

```diff
@@ -0,0 +1,692 @@
+                                        kCanGetMyAccountDetail}),
+                    kCanReceive)
+      ->sendTx(
+          accountGrantToAccount(kAccount1,
+                                kAccount1Keypair,
+                                kAccount2,
+                                permissions::Grantable::kSetMyAccountDetail))
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 1); })
+      .sendTx(permiteeSetAccountDetail(kAccount2,
+                                       kAccount2Keypair,
+                                       kAccount1,
+                                       kAccountDetailKey,
+                                       kAccountDetailValue))
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 1); })
+      .sendQuery(queryAccountDetail(kAccount1, kAccount1Keypair),
+                 check_account_detail)
+      .done();
+}
+
+/**
+ * C261 Grant transfer permission
+ * @given an account with rights to grant transfer of his/her assets
+ * AND the account can receive assets
+ * AND the account has some amount of assets
+ * AND the account has permitted to some other account in the system
+ * to transfer his/her assets
+ * @when the permitee transfers assets of the account
+ * @then a block with transaction to grant right is written
+ * AND the transfer is made
+ */
+TEST_F(GrantPermissionTest, DISABLED_GrantTransferPermission) {
+  auto amount_of_asset = "1000.00";
+
+  createTwoAccounts(concatenateVectors({kCanTransferMyAssets, kCanReceive}),
+                    concatenateVectors({kCanTransfer, kCanReceive}))
+      ->sendTx(accountGrantToAccount(kAccount1,
+                                     kAccount1Keypair,
+                                     kAccount2,
+                                     permissions::Grantable::kTransferMyAssets))
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 1); })
+      .sendTx(addAssetAndTransfer(IntegrationTestFramework::kAdminName,
+                                  kAdminKeypair,
+                                  amount_of_asset,
+                                  kAccount1))
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 1); })
+      .sendTx(transferAssetFromSource(
+          kAccount2, kAccount2Keypair, kAccount1, amount_of_asset, kAccount2))
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 1); })
+      .done();
+}
+
+/**
+ * C262 GrantPermission without such permissions
+ * @given an account !without! rights to grant rights to other accounts
+ * @when the account grants rights to an existing account
+ * @then this transaction is statefully invalid
+ * AND it is not written in the block
+ */
+TEST_F(GrantPermissionTest, GrantWithoutGrantPermissions) {
```

Comment:
should I expand this list to every grantable permission?

Reply:
@lebdron

Reply:
That would be nice. TYPED_TEST fits pretty good here, so you basically don't need to change much